compat/os/mingw.h

```diff
@@ -46,7 +46,9 @@ fsync(int fd)
 }
 #endif
 
-#define mkdir(p,F) mkdir((p))
+/* importaed from mswindows.h */
+#define mkdir(p, F) ::mkdir((p))
+#define pipe(pipefd) ::_pipe((pipefd), 4096, _O_BINARY)
```

Comment:
As in #2018, when duplicating code, please duplicate code as faithfully as possible (instead of adding a new variation). Please disclose and justify any necessary differences.

Reply:
As a general rule that might be good advice. However, these are portability definitions targeted at specific build environments. As such having differences in the code here are expected and often better - the code varies as the environment needs vary.

Reply:
> having differences in the code here are expected

I disagree that introducing differences is _expected_ when duplicating code. In exceptional cases, where differences are indeed necessary, they need to be justified (e.g., explaining why this "environment needs" are different and how that difference requires different code).

In this particular case, I doubt the differences are necessary, and I do not see any justification to introduce them.